feat(notification-badge): added expanded state

mcoker opened the pull request:
fixes #4664

patternfly-build commented:
Preview: https://patternfly-pr-5048.surge.sh

mcarrano reviewed (approved):
Looks good to me @mcoker . Do we need any follow up to start using this in out masthead demos and examples?

mcoker commented:
@mcarrano yep, that's probably a good idea. I can go through the core demos and make any updates needed in this PR, or we can make that a separate task. I already updated the notification drawer demos in this PR which _I think_ is all that needs to be updated. Then we'll need a react issue to do the same.

We may also consider removing the notification bell from the default masthead used in full-page demos, and only including it when clicking on it opens a notification drawer. Otherwise the basic full-page demo masthead would just include generic actions and basic functionality (a dropdown or whatever would trigger a menu, but with generic items).  WDYT?

mcarrano commented:
That all sounds good @mcoker . Maybe open follow-up issues and I'll pull them into a future milestone?

mattnolting commented on src/patternfly/components/NotificationBadge/examples/NotificationBadge.md:
````diff
@@ -52,6 +52,50 @@ import './NotificationBadge.css'
+
+```hbs
+<div class="pf-t-dark">
+{{#> button button--modifier="pf-m-plain" button--attribute='aria-label="Notifications" aria-expanded="true"'}}
````
Were you using `aria-expanded="true"` here instead of `button--IsExpanded="trye"` to avoid applying `pf-m-expanded` to the button itself?

mcoker replied:
Yeah, it's only supported on the control button, and noticed the notification badge button in the masthead was using the expanded class, so updated that just to be `aria-expanded`, too. Looking at the react component, I think we'll want to add that there, too - https://github.com/patternfly/patternfly-react/blob/main/packages/react-core/src/components/NotificationBadge/NotificationBadge.tsx

mattnolting commented on src/patternfly/components/NotificationBadge/notification-badge.scss:
```diff
@@ -108,9 +111,15 @@
 
     &:hover {
-      --pf-c-notification-badge--after--BackgroundColor: var(--pf-c-notification-badge--m-attention--hover--after--BackgroundColor);
+      --pf-c-notification-badge--m-attention--after--BackgroundColor: var(--pf-c-notification-badge--m-attention--hover--after--BackgroundColor);
```
👍

mattnolting reviewed (commented):
LGTM 👍 Just a couple comments.

srambach commented:
Just checking - is this the intended dark theme color (especially for the attention one)? Just seems off to me. 🤷 
<img width="164" alt="image" src="https://user-images.githubusercontent.com/19825616/187538845-b48a0d31-3b24-484b-b2da-284791e99c46.png">

mcoker commented:
@srambach good callouts. I have an issue here for the attention icon color - https://github.com/patternfly/patternfly/issues/5049. For now I've updated it to be black to match our danger button.

The read icon should be white (or whatever the hover color is) when expanded. Updated that, too!

srambach reviewed (approved):
LGTM 🔔

patternfly-build commented:
:tada: This PR is included in version 4.211.0 :tada:

The release is available on:
- [GitHub release](https://github.com/patternfly/patternfly/releases/tag/prerelease-v4.211.0)
- [npm package (@latest dist-tag)](https://www.npmjs.com/package/@patternfly/patternfly/v/4.211.0)

Your **[semantic-release](https://github.com/semantic-release/semantic-release)** bot :package::rocket:
